### PR TITLE
improvements for index creation

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Enterprise companion PR: https://github.com/arangodb/enterprise/pull/734
Partially addresses https://arangodb.atlassian.net/browse/BTS-545

Small improvements for index creation:

* Preselect "create index in background" option when creating indexes in
  the web UI. The "create index in background" option can be less intrusive
  because it allows other write operations on the collection to proceed.

* Do not block a scheduler thread on the coordinator while an index is being 
  created. Instead, start a background thread for being the actual index 
  fill-up work. The original thread can then be relinquished until the index
  is completely filled or index creation has failed. This also allows
  obsoleting the startup option `--cluster.index-create-timeout`, which from
  now on is ignored when set.

- [x] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [x] :book: CHANGELOG entry made

#### Backports:

- [x] Backports required for: *3.8*, *3.7*

#### Related Information

- [x] Enterprise PR: https://github.com/arangodb/enterprise/pull/734

### Testing & Verification

- [x] The behavior in this PR was *manually tested*
- [x] This change is already covered by existing tests, such as *shell_client, http_server, shell_server, shell_server_aql, audit_client*.

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -31,7 +42,11 @@ devel
 * Fixed: /_api/transaction/begin called on edge collections of disjoint
   SmartGraphs falsely returned CollectionNotFound errors.
 
+<<<<<<< HEAD
+* Bug-Fix: In more complex queries there was a code-path where a (Disjoint-) 
+=======
 * Bugfix: In more complex queries there was a code-path where a (Disjoint-) 
+>>>>>>> 5eaf1d23514db50d4b0403f4f82a293ff89d42bf
```
Comment (jsteemann) [inline CHANGELOG:49]:
```suggestion
* Bugfix: In more complex queries there was a code-path where a (Disjoint-) 
```

**Review by neunhoef — APPROVED** [review]
LGTM

File: arangod/Utils/Events.h [hunk 2]
```diff
@@ -57,8 +57,10 @@ void TruncateCollection(std::string const& db, std::string const& name,
                         OperationResult const& result);
 void CreateDatabase(std::string const& name, Result const& result, ExecContext const& context);
 void DropDatabase(std::string const& name, Result const& result, ExecContext const& context);
-void CreateIndex(std::string const& db, std::string const& col,
-                 VPackSlice const& slice, ErrorCode result);
+// index creation start
+void CreateIndex(std::string const& db, std::string const& col, VPackSlice slice);
+// index creation result
+void CreateIndex(std::string const& db, std::string const& col, VPackSlice slice, ErrorCode result);
```
Comment (mpoeter) [inline arangod/Utils/Events.h:63]: Please rename these to make it more clear, e.g., `CreateIndexBegin` and `CreateIndexEnd` or something like that.

**Review by mpoeter — CHANGES_REQUESTED** [review]

**Comment by mpoeter** [comment]
One comment, otherwise LGTM.

File: CHANGELOG [hunk 3]
```diff
@@ -1,6 +1,23 @@
 devel
 -----
 
+* Preselect "create index in background" option when creating indexes in
+  the web UI. The "create index in background" option can be less intrusive
+  because it allows other write operations on the collection to proceed.
+
+* Do not block a scheduler thread on the coordinator while an index is being 
+  created. Instead, start a background thread for being the actual index
```
Comment (mpoeter) [inline CHANGELOG:9]:
```suggestion
  created. Instead, start a background thread for the actual index 
```

File: arangod/Utils/Events.h (hunk 2 above)
Comment (jsteemann) [reply arangod/Utils/Events.h]: Ok, renamed.

**Review by mpoeter — APPROVED** [review]

**Pull request merged**